### ISSUE TITLE
Use more stringent prefix filtering

### ISSUE THREAD

**Issue by caufieldjh** [monarch-initiative/phenio#8]
Imported ontologies may contain any number of prefixes, including ones conflicting with other sources.
We should limit sources to expected ones to avoid downstream node conflicts.

**Comment by caufieldjh** [comment]
Pay specific attention to HGNC prefixes, which take multiple forms.
In some cases, we may not want to bring in the version with all imports but rather the base version

**Comment by matentzn** [comment]
"Imported ontologies may contain any number of prefixes" - just to be clear here: ontologies do not, apart from the exceptions of hasDbXref values, contain _any_ prefixes. They may use diverging URI bases for various ID schemes like HGNC. How do you intend to harmonise that? SPARQL update?

**Comment by caufieldjh** [comment]
True - I'm using "prefixes" here to mean "namespaces", so the issue is just that we're using a bunch of basefiles and don't have any control over which namespaces are involved there. So in phenio, downstream users (for now, mostly just Monarch) may want to ingest OMIM but then find that phenio is including a bunch of OMIM IDs through mondo, both as CURIEs and as IDs. The easy step for them is to just filter out all of OMIM from phenio, but we can make that more feasible, at least.

I'm tempted to think that some SPARQL could consolidate the URIs used by HGNC, OMIM, etc., but I expect that will require more comprehensive knowledge of the types of axioms they're used in. The hasDbXref values have CURIE prefixes, like you say, but so do oboInOwl:source values for datatypes. Maybe that's it though, and I honestly don't know what happens to those source values in KGX.

**Comment by caufieldjh** [comment]
Unless there's another way to handle this in robot, like checking the output of robot measure

**Comment by matentzn** [comment]
The way I would have like to solve this would be:

1. URI in
2. use bioregistry to find prefix (using the biolink context as preferred context)
3. create CURIE
4. use bioregistry to lookup preferred expansion
5. CURIE to URI with new prefix map

This could actually be handled in python, and maybe even something like OAK.

**Comment by matentzn** [comment]
Did you mean to close this issue?

**Comment by caufieldjh** [comment]
No, I had detached it from the linked PR but it looks like that didn't take
